src/prefect/tasks/prefect/flow_run.py

```diff
@@ -169,7 +170,11 @@ def create_flow_run(
 
 @task
 def get_task_run_result(
-    flow_run_id: str, task_slug: str, map_index: int = -1, poll_time: int = 5
+    flow_run_id: str,
+    task_slug: str,
+    map_index: int = -1,
+    poll_time: int = 5,
+    serializer: Serializer = None,
```

Comment:
maybe a bad question because I'm not too familiar with type hints, but should this be `Optional[Serializer]`? or, when are we supposed to use a type `T` vs `Optional[T]`?

Reply:
Assignment of default values to `None` makes the inferred type automatically `Optional`. There's been a recent movement to avoid this pattern (I can't find a great source covering it right now), but we use it everywhere still. Eventually, we'll make the shift to explicit `Optional` types but they've also shifted the standard pattern to `T | None` in [PEP-0604](https://peps.python.org/pep-0604/) in Python 3.10+.

Reply:
https://mypy.readthedocs.io/en/stable/command_line.html#none-and-optional-handling

Reply:
Ooo, awesome. Thanks for the education & links!